src/clusterfuzz/_internal/datastore/data_handler.py

```diff
@@ -110,13 +110,20 @@ def get_domain():
 
 
 def get_testcase_by_id(testcase_id):
-  """Return the testcase with the given id, or None if it does not exist."""
-  if not testcase_id or not str(testcase_id).isdigit() or int(testcase_id) == 0:
-    raise errors.InvalidTestcaseError
+  """Return the testcase with the given id.
+  Raises InvalidTestcaseError if no such testcase exists.
+  """
+  try:
+    parsed_id = int(testcase_id)
+  except ValueError:
+    raise errors.InvalidTestcaseError(testcase_id)
+
+  if parsed_id == 0:
```

Comment:
not so sure this is necessary.

Reply:
Agreed. We might want to check for non-positive values in general, but the ndb lookup should fail anyway if the value is wrong. It seems just a bit faster to check locally first.

Reply:
Yeah, I probably wouldn't have included this check but I guess since it was already there.